### PR TITLE
Improve the user interaction for add/remove root keys

### PR DESCRIPTION
**Pull request by kairoaraujo**
# Description

This PR improves the usability of the CLI while interacting with the Root Keys, making the interface easier to use for add, remove root keys.

https://github.com/repository-service-tuf/repository-service-tuf-cli/assets/546882/fa6ba2b3-fb6a-4b96-8789-538f88a86928

# Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

# Additional requirements
- [x] Tests have been added for the bug fix or new feature
- [] Docs have been added for the bug fix or new feature

# Code of Conduct
By submitting this PR, you agree to follow our [Code of Conduct](CODE_OF_CONDUCT.rst).

- [x] I agree to follow this project's Code of Conduct

### CODE REVIEW

**Comment by kairoaraujo** [comment]
I need to figure out how to implement it for the FT.

File: repository_service_tuf/cli/admin/helpers.py [hunk 1]
```diff
@@ -405,17 +377,24 @@ def _add_root_signatures_prompt(
...
-            len(keys), allow_skip=bool(root_result.signed)
-        )
-        if choice == -1:
+        key_choices = {
+            key.unrecognized_fields.get(KEY_NAME_FIELD, key.keyid): key
+            for key in keys
+        }
+        sign_options = list(key_choices.keys())
+        or_continue = ":"
+        if bool(root_result.signed):
```
Comment (MVrachev) [inline repository_service_tuf/cli/admin/helpers.py:386]: Interesting trick with the cast to bool, but I am not sure it's better than just checking if `len((root_result.signed) > 0`.
I think by using length you make it clearer what is the intention - check whether there is at least one key that  signed root.

Or will it probably be even clearer to check `len(root_md.signatures) > 1`?

File: repository_service_tuf/cli/admin/helpers.py [hunk 2]
```diff
@@ -405,17 +377,24 @@ def _add_root_signatures_prompt(
...
+            or_continue = " or continue:"
+            sign_options.insert(0, "continue")
+
+        console.print(f"\nSelect one current key to sign{or_continue}")
```
Comment (MVrachev) [inline repository_service_tuf/cli/admin/helpers.py:390]: Seems like this sentence could be better formulated.
What about:
`f"\nSelect a key for signing{or_continue}"`

File: repository_service_tuf/cli/admin/helpers.py [hunk 3]
```diff
@@ -405,17 +377,24 @@ def _add_root_signatures_prompt(
...
+        choice = _select(sign_options)
+
+        if choice == "continue":
+            console.print("\nSelect one current key for sign")
```
Comment (MVrachev) [inline repository_service_tuf/cli/admin/helpers.py:394]: Why do we print that when we would exit this function?

File: tests/conftest.py [hunk 4]
```diff
@@ -153,29 +153,45 @@ def ceremony_inputs() -> Tuple[List[str], List[str], List[str], List[str]]:
...
         f"{_PEMS / 'JH.ed25519'}",  # Please enter path to encrypted private key  # noqa
-        "1",  # Please enter signing key index
         f"{_PEMS / 'JJ.ecdsa'}",  # Please enter path to encrypted private key  # noqa
     ]
 
     return input_step1, input_step2, input_step3, input_step4
 
 
+@pytest.fixture
+def ceremony_selection(monkeypatch) -> lambda *a: str:
```
Comment (MVrachev) [inline tests/conftest.py:174]: Not needed:
```suggestion
def ceremony_selection() -> lambda *a: str:
```

File: tests/conftest.py [hunk 5]
```diff
@@ -153,29 +153,45 @@ def ceremony_inputs() -> Tuple[List[str], List[str], List[str], List[str]]:
         "2",  # Please enter root threshold
         f"{_PEMS / 'JC.pub'}",  # Please enter path to public key
         "my rsa key",  # Please enter key name
-        "0",  # Please press 0 to add key, or remove key by entering its index  # noqa
```
Comment (MVrachev) [inline tests/conftest.py:156]: I think it will be good to have a comment in the beginning of the `ceremony_inputs` fixture mentioning that key addition, removal and signing for `input_step_2` and `input_step4` are mocked using `ceremony_selection` fixture.

File: tests/conftest.py [hunk 6]
```diff
@@ -153,29 +153,45 @@ def ceremony_inputs() -> Tuple[List[str], List[str], List[str], List[str]]:
...
 
     return input_step1, input_step2, input_step3, input_step4
 
 
+@pytest.fixture
+def ceremony_selection(monkeypatch) -> lambda *a: str:
+    # public key selection options
+    selection_options = iter(
+        (
+            # adding/removing root public signing keys
```
Comment (MVrachev) [inline tests/conftest.py:178]: Besides this comment probably be worth it to map it to `input_steps` from `ceremony_inputs` like that:
```suggestion
            # adding/removing root public signing keys (selections for input_step2)
```

File: tests/conftest.py [hunk 7]
```diff
@@ -153,29 +153,45 @@ def ceremony_inputs() -> Tuple[List[str], List[str], List[str], List[str]]:
...
+            "add",  # add key
+            "add",  # add key
+            "remove",  # remove key
+            "my rsa key",  # select key to remove
+            "continue",  # continue
+            # signing with root keys
```
Comment (MVrachev) [inline tests/conftest.py:184]: Same here like:
```suggestion
            # signing with root keys (selections for input_step4)
```

File: tests/unit/cli/admin/test_ceremony.py [hunk 8]
```diff
@@ -9,12 +9,21 @@
...
+        monkeypatch,
         ceremony_inputs,
+        ceremony_selection,
         client,
         test_context,
         patch_getpass,
         patch_utcnow,
     ):
+
+        # public key selection options
```
Comment (MVrachev) [inline tests/unit/cli/admin/test_ceremony.py:21]:
```suggestion
        # public keys and signing keys selection options
```

File: tests/unit/cli/admin/test_ceremony.py [hunk 9]
```diff
@@ -9,12 +9,21 @@
...
+        monkeypatch.setattr(
+            "repository_service_tuf.cli.admin.helpers._select",
```
Comment (MVrachev) [inline tests/unit/cli/admin/test_ceremony.py:23]: Make `repository_service_tuf.cli.admin.helpers` as one constant in this file and shorten this to a one line like I did here:
https://github.com/repository-service-tuf/repository-service-tuf-api/blob/6eee4eb83a19e3b7c38171e92ffcb39de1b53979/tests/unit/api/test_bootstrap.py#L14C1-L14C10
it will improve readability and shorten changes.

File: tests/unit/cli/admin/test_ceremony.py [hunk 10]
```diff
@@ -42,16 +56,18 @@ def test_ceremony_threshold_less_than_2(
...
             f"{_PEMS / 'JH.pub'}",  # Please enter path to public key
             "JimiHendrix's Key",  # Please enter key name
-            "0",  # Please press 0 to add key, or remove key by entering its index.  # noqa
             f"{_PEMS / 'JJ.pub'}",  # Please enter path to public key
             "JanisJoplin's Key",  # Please enter key name
-            "1",  # Please press 0 to add key, or remove key by entering its index. Press enter to contiue  # noqa
-            "",  # Please press 0 to add key, or remove key by entering its index. Press enter to contiue  # noqa
         ]
 
+        # public key selection options
```
Comment (MVrachev) [inline tests/unit/cli/admin/test_ceremony.py:65]:
```suggestion
        # public keys and signing keys selection options
```

File: tests/unit/cli/admin/test_ceremony.py [hunk 11]
```diff
@@ -83,6 +104,11 @@ def test_ceremony__non_positive_expiration(
             "",  # Please enter days until expiry for root role (365)
         ]
```
Comment (MVrachev) [inline tests/unit/cli/admin/test_ceremony.py:106]: Will you add the comment or not:
```suggestion
        # public keys and signing keys selection options
```

File: tests/unit/cli/admin/test_ceremony.py [hunk 12]
```diff
@@ -212,39 +256,13 @@ def test_ceremony_online_key_one_of_root_keys(
...
-        with open(_PAYLOADS / "ceremony.json") as f:
-            expected = json.load(f)
-
-        sigs_r = result.data["metadata"]["root"].pop("signatures")
-        sigs_e = expected["metadata"]["root"].pop("signatures")
-
-        assert [s["keyid"] for s in sigs_r] == [s["keyid"] for s in sigs_e]
-        assert result.data == expected
-        assert "Key already in use." in result.stdout
+        # public key selection options
```
Comment (MVrachev) [inline tests/unit/cli/admin/test_ceremony.py:260]:
```suggestion
        # public keys and signing keys selection options
```

File: tests/unit/cli/admin/test_sign.py [hunk 13]
```diff
@@ -14,11 +14,18 @@
...
+    def test_sign_with_previous_root(
+        self, monkeypatch, test_context, patch_getpass
+    ):
         inputs = [
-            "1",  # Please enter signing key index
             f"{_PEMS / 'JH.ed25519'}",  # Please enter path to encrypted private key  # noqa
         ]
+        # selections interface
+        monkeypatch.setattr(
+            "repository_service_tuf.cli.admin.helpers._select",
```
Comment (MVrachev) [inline tests/unit/cli/admin/test_sign.py:25]: Use https://github.com/repository-service-tuf/repository-service-tuf-cli/blob/bfbf0a545024d9e72c533bb2a7318c26e5e7a123/tests/conftest.py#L45 to shorten paths.

File: tests/unit/cli/admin/test_update.py [hunk 14]
```diff
@@ -4,32 +4,51 @@
...
             f"{_PEMS / 'JJ.ecdsa'}",  # Please enter path to encrypted private key  # noqa
-            "1",  # Please enter signing key index
             f"{_PEMS / 'JC.rsa'}",  # Please enter path to encrypted private key  # noqa
         ]
         args = [f"{_ROOTS / 'v1.json'}"]
 
+        # selections interface
+        selection_options = iter(
+            (
+                # adding/removing root public signing keys
```
Comment (MVrachev) [inline tests/unit/cli/admin/test_update.py:32]: It's better to add same clarifycations as I mentioned for `ceremony_selection`.

**Review by MVrachev — CHANGES_REQUESTED** [review]
The result is great!
My comments are more cosmetic improvements for clarity, readability and shortness.

I see you removed:
`test_ceremony_try_setting_root_keys_less_than_threshold(`
Given that we use `selection` you think there is no way for this to happen?

File: .github/workflows/ci.yml [hunk 15]
```diff
@@ -12,7 +12,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-versions: [ "3.9", "3.10", "3.11", "3.12" ]
+        python-versions: [ "3.10", "3.11", "3.12" ]
```
[the remark this answers is not in the document]
Comment (MVrachev) [reply .github/workflows/ci.yml:15]: It's because it's a CLI and we don't know the users local python version.

File: repository_service_tuf/cli/admin/helpers.py (hunk 1 above)
Comment (kairoaraujo) [reply repository_service_tuf/cli/admin/helpers.py:386]: We only want to check if the initial signature is present.
I reused this code from the previous logic.

File: repository_service_tuf/cli/admin/helpers.py [hunk 16]
```diff
@@ -4,8 +4,9 @@
 
 from dataclasses import dataclass
 from datetime import datetime, timedelta, timezone
-from typing import Any, Dict, Optional, Tuple
+from typing import Any, Dict, List, Optional, Tuple
 
+import beaupy  # type: ignore
```
[the remark this answers is not in the document]
Comment (kairoaraujo) [reply repository_service_tuf/cli/admin/helpers.py:9]: beaupy doesn't have a typing
```
typing: commands[0]> mypy --exclude tests/ --exclude rstuf-umbrella/ .
repository_service_tuf/cli/admin/helpers.py:9: error: Skipping analyzing "beaupy": module is installed, but missing library stubs or py.typed marker  [import-untyped]
```

**Review by MVrachev — APPROVED** [review]
